### ISSUE TITLE
Theme: 18.30 after update issue on CentOS 6.8

### ISSUE THREAD

**Issue by matrix200** [qooob/authentic-theme#592]
Before posting an issue, please search through README.md and closed/opened issues to make sure that the answer to your questions hasn't been answered/discussed already.

Please make sure that you have latest theme (18.30) and Webmin (1.820) versions installed.

Provide the following information:
* What is your server-side OS and its version?
* What is your client-side OS, browser and its versions?
* What are the steps to reproduce the issue

**Comment by qooob** [comment]
Like what breaks?

Did you restart Webmin using console?

**Comment by qooob** [comment]
It's available here:

https://github.com/qooob/authentic-theme/releases/tag/18.20

**Comment by matrix200** [comment]
Tried several times to restart webmin.
It just never loads.
After reverting to another theme (it required forced reinstall of webmin) all links appear but clicking them does nothing.

**Comment by matrix200** [comment]
Thanks.
Will try to install now and report back.

**Comment by qooob** [comment]
That sounds odd. Can you provide a screenshot? Any errors in console?

**Comment by qooob** [comment]
I have it tested on CentOS 6.8 and Debian 7

**Comment by qooob** [comment]
I'm running 18.30 in production and on 2 debug servers. All working fine. Tried updating from 18.20 to 18.30. For Webmin everything works great.

However Usermin will need manual update again. Will fix it upon next release.

**Comment by matrix200** [comment]
Yep it works.
The 18.20 that is.

**Comment by qooob** [comment]
Try 18.30 again. It just should work.

**Comment by matrix200** [comment]
Should I try the regular update?
Through the gui?

**Comment by matrix200** [comment]
Or use the gzip?

**Comment by qooob** [comment]
Yes, using GUI would be alright as well.

**Comment by matrix200** [comment]
OK let me try again.

**Comment by matrix200** [comment]
So far same behavior :(
The update proceeds very quickly to almost 95 (the visual bar) and then its stuck and is just spinning there.
Just like the last time

**Comment by matrix200** [comment]
The last time after waiting for a few minutes I decided to restart the webmin after which it came up completely broken

**Comment by matrix200** [comment]
Perhaps I should wait longer?

**Comment by matrix200** [comment]
Its broken again

**Comment by matrix200** [comment]
OK now I know how to restore it at least :)

**Comment by qooob** [comment]
Ok, :)

I just tested it in CentOS 6.8. Yes it happens but all you need is to hit reload button in browser or just goto browser's URL and hit enter.

**Comment by matrix200** [comment]
I tried it
It just spins there forever and after webmin restart its just dead

**Comment by matrix200** [comment]
Reload or trying to open it in another browser tab just does nothing

**Comment by qooob** [comment]
It's your browser. What browser are you using?

Try incognito mode. What authentication mechanism are you using?

**Comment by matrix200** [comment]
Tried both Palemoon 27.0.2 (Mozilla based) and Ie11

**Comment by matrix200** [comment]
Using session auth (no two factor setup)
Also tried clearing cookies/cache/temp files

**Comment by qooob** [comment]
Non of those are supported! ;) 

Clear browser's cache!

Recommend latest Firefox or Chrome.

Open console and see for errors. There should be some errors is there are some major changes.

Try restarting your browser. If doesn't help. Local machine and then server. It'll work, I bet!! Try totally different computer.

**Comment by matrix200** [comment]
Well Palemoon should be kind of supported, no?
I mean its firefox based (I believe on firefox 50).
I don't use either chrome or firefox (both memory hogs imo).
Can't really restart server as its pretty critical server (its my smart home hub).

**Comment by qooob** [comment]
Please try installing Chrome/Chromium and test it there (out of curiosity).

Yes, Firefox 50 was tested and worked fine as well.

**Comment by qooob** [comment]
Try `Ctrl+Shift+R` to hard reload the page.

**Comment by matrix200** [comment]
I believe I have portable chrome somewhere, I will try to get it.
Funny detail though.
After it fails, there is another pane/tab on the right side of the screen where it shows all system details, but it shows all of them to be null.
And the progress wheel is spinning indefintely in the middle.
Clicking on all the modules does nothing.
Also it displays an annoying messages (do you want to leave this page?) if any link is clicked, or if an attempt is made to close its tab.

**Comment by qooob** [comment]
No worries!! 

Just dump browsers cache!! Just tested it in Firefox 50.0.2 - it works great!

**Comment by matrix200** [comment]
ok I will try it tomorrow :)
Its almost midnight here.

**Comment by itmustbe** [comment]
I had this same issue and it really scared me briefly (feeling locked out of my server with the endless looping loading graphic in the center of the screen), not knowing how to change to a working theme! Switching from Safari with Private Browsing to Chrome fixed it for me for the moment :)  Really scared me though because logging into my server with Webmin has been working just fine in Safari with Private Browsing, right up until I clicked the button to update the Authentic theme from 18.20 to 18.30 (tried logging out and then back in too using Safari).  In my Dev Tools inspector the errors were all of this form:  [Error] QuotaExceededError (DOM Exception 22): The quota has been exceeded. (in init.min.js and parent.min.js)  I'm on CentOS 6.8 with Webmin 1.821

**Comment by qooob** [comment]
..but now it's fine after browser restart/clearing cache, right?

**Comment by itmustbe** [comment]
On further testing 18.30 works in Safari so long as I am not using Private Browsing mode.  In Private Browsing mode, the two sidebars show up, but never anything more than an endless blue spinner in the main content area.  This is in contrast to 18.20 working great in Safari using Private Browsing mode.  Attached are the console errors when using Private Browsing mode:
<img width="466" alt="screen shot 2016-12-07 at 10 16 00 pm" src="https://cloud.githubusercontent.com/assets/200762/20999721/80c43024-bccb-11e6-8533-8ebead6d694f.png">

**Comment by qooob** [comment]
Just run in console `localStorage.clear()`;

**Comment by qooob** [comment]
http://stackoverflow.com/questions/14555347/html5-localstorage-error-with-safari-quota-exceeded-err-dom-exception-22-an

It's intended by Safari in private mode browsing.

**Comment by qooob** [comment]
It will be fixed in 18.31

**Comment by qooob** [comment]
18.31 is out, please upgrade.

**Comment by itmustbe** [comment]
That was fast!  Works great again with 18.31, thanks so much!

**Comment by qooob** [comment]
You're welcome. It wasn't fast, really but thanks!

**Comment by matrix200** [comment]
Not for me.
Still the same error screen.

**Comment by qooob** [comment]
1. Make sure that the theme has been updated
2. Clear browser's cache.
3. Restart the browser, then Webmin/Usermin

In case it's not solved, provide the screenshot with console opened to see an error.

**Comment by matrix200** [comment]
Did all that.
Same error, same screen, that Elise has shared.
Also tried incognito mode with the same result.

**Comment by qooob** [comment]
:)

Provide extra details. Otherwise I can't help.

**Comment by matrix200** [comment]
OK I will try to do some more testing tomorrow, its midnight here and I really need to hit the sack.
